Merge/foundation release/1.10.26

ziogaschr opened the pull request:
This merges go-ethereum tag v1.10.26 over our latest https://github.com/etclabscore/core-geth/tree/merge/foundation-release/1.10.23-tests-generate-fixes-statefix-difficulty-test-v2-ripemd-tx-debug-no-ripemd-hack-evmc-dimitrys-etc-tests

ziogaschr committed:
cmd/geth: fix on darwin, where long test names fail

Error is "The ipc endpoint is longer than %d characters."

ziogaschr committed:
go.mod,go.sum: update dependency for etclabscore/go-openrpc-reflect@fix-m1-nil-pointer for fixing test issue

ziogaschr committed:
cmd/geth: fix lint issue (G401: Use of weak cryptographic primitive (gosec))

meowsbits committed:
cmd/geth: fixup subtest names, logging, and datadir paths

If IPC paths are too long, Unix systems will complain
because there are limits, albeit inconsistent ones.
So we try to keep IPC path names short (ie <100 chars).

I removed all instances where flags were used
in test names because
- they cause issues with apple silicon M1,
- and they are pointless

I also removed the redundant subtest naming scheme
which included the test name. Go includes the
test name by default anyway.

Date: 2022-08-29 09:21:05-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
go.mod,go.sum: bump etclabscore/go-openrpc-reflect to v0.0.37

Bumps dep to a tag version equivalent to the
previously referenced, untagged version.

Date: 2022-08-29 09:22:26-05:00
Signed-off-by: meows <b5c6@protonmail.com>

s1na committed:
graphql: return correct logs for tx (#25612)

* graphql: fix tx logs

* minor

* Use optimized search for selecting tx logs

ziogaschr committed:
Merge pull request #498 from etclabscore/fix/tests-mac

Fix tests on MacOS (darwin)

holiman committed:
eth/protocols/snap: fix problems due to idle-but-busy peers (#25651)

meowsbits committed:
consensus/ethash: apply new cache file naming pattern to full datasets too

Date: 2022-08-31 19:55:29-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge pull request #491 from etclabscore/merge/foundation-release/1.10.21

Merge/foundation release/1.10.21

meowsbits committed:
Merge pull request #496 from etclabscore/merge/foundation-release/1.10.21-tests-generate

Merge/foundation release/1.10.21 tests generate

meowsbits committed:
Merge pull request #499 from etclabscore/merge/foundation-release/1.10.21-tests-generate-etchash-mordor-pr-2

fix Etchash cache at Mordor

karalabe committed:
trie: check childrens' existence concurrently for snap heal (#25694)

karalabe committed:
eth/protocols/snap: throttle trie heal requests when peers DoS us (#25666)

* eth/protocols/snap: throttle trie heal requests when peers DoS us

* eth/protocols/snap: lower heal throttle log to debug

Co-authored-by: Martin Holst Swende <martin@swende.se>

* eth/protocols/snap: fix comment

Co-authored-by: Martin Holst Swende <martin@swende.se>

s1na committed:
graphql: fixes missing tx logs (#25745)

* graphql: fix tx logs

* graphql: refactor test service setup

* graphql: add test for tx logs

karalabe committed:
Release Geth v1.10.24

MariusVanDerWijden committed:
params: set TerminalTotalDifficultyPassed to true (#25769)

* params: set TerminalTotalDifficultyPassed to true

* Update params/config.go

Co-authored-by: Martin Holst Swende <martin@swende.se>

karalabe committed:
params: release Geth v1.10.25

karalabe committed:
eth: fix a rare datarace on CHT challenge reply / shutdown (#25831)

dependabot[bot] committed:
build(deps): bump joblib from 1.0.0 to 1.2.0

Bumps [joblib](https://github.com/joblib/joblib) from 1.0.0 to 1.2.0.
- [Release notes](https://github.com/joblib/joblib/releases)
- [Changelog](https://github.com/joblib/joblib/blob/master/CHANGES.rst)
- [Commits](https://github.com/joblib/joblib/compare/1.0.0...1.2.0)

---
updated-dependencies:
- dependency-name: joblib
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

meowsbits committed:
Merge pull request #503 from etclabscore/dependabot/pip/joblib-1.2.0

build(deps): bump joblib from 1.0.0 to 1.2.0

meowsbits committed:
:hammer:

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: change ETC test fork configs to chainid=61 (not =1)

Date: 2022-10-25 12:22:04-07:00
Signed-off-by: meows <b5c6@protonmail.com>

holiman committed:
eth/filters: change filter block to be by-ref (#26054)

This PR changes the block field in the filter to be a pointer, to disambiguate between empty hash and no hash

meowsbits committed:
tests: generated state tests

Date: 2022-10-28 09:22:06-07:00
Signed-off-by: meows <b5c6@protonmail.com>

jmank88 committed:
rpc: handle wrong HTTP batch response length (#26064)

fjl committed:
params: release geth v1.10.26 stable

theshoe1029 committed:
add custom interpreter to t8n evm config

meowsbits committed:
cmd/evm: use utils.EVMInterpreterFlag for all evm tool apps

Date: 2022-11-07 10:03:55-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/evm/internal/t8ntool,cmd/evm: make sure the vm inits the external interpreter when set

Date: 2022-11-07 10:04:36-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/utils: initialize external evms if configured for geth

Date: 2022-11-07 10:17:54-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm,go.mod,go.sum: bump ethereum/evmc version: v7.5.0 -> v10.0.0

Date: 2022-11-07 10:43:54-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: implement hostContext.AccessAccount,.AccessStorage

Making sure that EIP2929 is enabled for current
context.

Date: 2022-11-07 11:10:34-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: update evmc.Storage* vars to new counterparts

For reference: https://github.com/ethereum/evmc/commit/35112aed7b4e0fd2f54cf2e3272ba9117a0c6494#diff-d1775ee3911821f864ea729e4ac4a459a522e37a604a5a67e07def298c75fb99R23

Date: 2022-11-07 11:46:32-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: evmc.TxContext no longer has Difficulty field

Date: 2022-11-07 11:47:09-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: return evmc.Berlin in case EIP2565 (as feature indicator) is enabled

Date: 2022-11-07 11:48:10-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: another syntax improvement for a comment

Date: 2022-11-07 11:48:49-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: update hostContext.Selfdestruct to return bool

Date: 2022-11-07 12:04:07-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: add gasRefund return value to hostContext.Call signature

Date: 2022-11-07 12:04:42-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: bump getRevision switch to support London

Also fixes wrong number of params argued.

Date: 2022-11-07 12:05:16-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: implement codeAddress argument for hostContext.Call sig

See https://github.com/ethereum/evmc/commit/8314761222c837d41f573b0af1152ce1c9895a32#diff-4c54ef259154e3cd3bd18b1963e027655525f909a887700cc2d1386e075c3628R144-R155

Date: 2022-11-07 12:12:44-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
build: bump evmone version: v0.5.0 -> v0.9.1

Date: 2022-11-07 12:52:10-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile,build: remove SSVM interpreter; project no longer exists

Original https://github.com/second-state/SSVM redirects
to https://github.com/WasmEdge/WasmEdge.

Date: 2022-11-07 13:15:09-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
build: bump hera version: v0.3.2 -> v0.6.0

v0.6.0 supports EVMC v10.

Date: 2022-11-07 14:05:21-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile,build: remove aleth interpreter

The ethereum/aleth project has been archived and
is no longer maintained.

This commit also removes an accidental leftover
from the ssvm code (call to clean in Makefile).

Date: 2022-11-07 14:07:59-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
.github/workflows: fixing c++/cmake issues w/r/t evmone, evmc tests

The command 'make test-evmc' complains about a GLIBCXX version
incompatibility (missing 3.4.29).

This issue has been shown resolved by the Actions CI run here:
https://github.com/meowsbits/core-geth/actions/runs/3429882085/jobs/5716103415
The run ultimately fails, but it fails because the tests fail,
while the evmone .so is used appropriately.

Full context of the error follows.

panic: EVMC loading error: /lib/x86_64-linux-gnu/libstdc++.so.6: version GLIBCXX_3.4.29 not found, required by /home/ia/go/src/github.com/ethereum/go-ethereum/build/_workspace/evmone/lib/libevmone.so

 goroutine 1 [running]:
 github.com/ethereum/go-ethereum/core/vm.initEVMC(0x11e4e0?, {0x7ffc235f574c?, 0xc0001361c0?})
       /home/ia/go/src/github.com/ethereum/go-ethereum/core/vm/evmc.go:83 +0x605
 github.com/ethereum/go-ethereum/core/vm.InitEVMCEVM({0x7ffc235f574c, 0x58})
       /home/ia/go/src/github.com/ethereum/go-ethereum/core/vm/evmc.go:60 +0xd0
 github.com/ethereum/go-ethereum/tests.TestMain(0xffffffffffffffff?)
       /home/ia/go/src/github.com/ethereum/go-ethereum/tests/init_test.go:50 +0xec
 main.main()
       _testmain.go:77 +0x1d3
 FAIL  github.com/ethereum/go-ethereum/tests   0.014s
 FAIL
 make: *** [Makefile:57: test-evmc] Error 1

See also here: https://github.com/etclabscore/core-geth/actions/runs/3414467176/jobs/5682462570#step:5:482

Date: 2022-11-09 07:44:38-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: use evmc.Revision values for logical conditions

I think this provides better readability, and
encourages readers to check and use the getRevision
both as a logical nexus, as well as to appreciate its
limitations.

Date: 2022-11-09 08:59:00-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge branch 'evmc-v10-fix-cmake' into evmc-v10-isaac

meowsbits committed:
.github/workflows: rename actions job to reflect not-just-EWASM

Date: 2022-11-09 09:10:19-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: fixup comment

Date: 2022-11-09 11:07:07-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: remove extra condition EIP2200 for Istanbul revision

While true, I'm debugging failing tests (when running
with --vm.evm=path/to/evmone.so), and want to remove
any possibly spurious logic.

Date: 2022-11-09 11:17:42-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: implement tx BaseFee and PrevRandao in case London enabled

Date: 2022-11-09 11:55:44-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: assign the refund value based on London rules... ?

Date: 2022-11-09 11:56:23-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
.github/workflows: fix directory path for git safe config for submodules

Date: 2022-11-09 11:57:19-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmake.sh: remove anachronstic development cmake.sh installation script, oops

Date: 2022-11-09 11:57:55-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge branch 'evmc-v10-fix-cmake' into evmc-v10-isaac

meowsbits committed:
Merge branch 'evmc-v10-isaac' into merge/foundation-release/1.10.23-tests-generate-fixes-statefix-difficulty-test-v2-ripemd-tx-debug-no-ripemd-hack-evmc

 Conflicts:
	cmd/evm/staterunner.go

meowsbits committed:
eth/ethconfig: run 'go generate ./...'

Its been a while...

Date: 2022-11-11 07:10:13-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core,params/types/genesisT: move 'go generate' decl to appropriate package

Date: 2022-11-11 07:14:37-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/genesisT: go generate, change is inoperative except deps tweaks

Date: 2022-11-11 07:28:57-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/evm/internal/t8ntool: add missing BaseFee field to ExecutionResult struct

And carry this through to assignment at the Apply method

This solves what felt like a big issue using core-geth's evm
with retesteth, which kept complaining as:

WARNING: tool vs retesteth basefee disagree: 0 vs 10
makeRPCError Invalid block: base fee not correct! Expected: '10', got: '0'
Finishing retesteth run
*** Total Tests Run: 0

Error: Invalid block: base fee not correct! Expected: '10', got: '0' (stRandom/randomStatetest153, fork: London, TrInfo: d: 0, g: 0, v: 0, TrData: )

--------
TestOutputHelper detected 1 errors during test execution!
/home/ia/dev/ethereum/retesteth/retesteth/TestOutputHelper.cpp(222): error: in 'GeneralStateTests/stRandom'

Date: 2022-11-11 08:02:47-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/evm/internal/t8ntool: add missing BaseFee field to ExecutionResult struct

And carry this through to assignment at the Apply method

This solves what felt like a big issue using core-geth's evm
with retesteth, which kept complaining as:

WARNING: tool vs retesteth basefee disagree: 0 vs 10
makeRPCError Invalid block: base fee not correct! Expected: '10', got: '0'
Finishing retesteth run
*** Total Tests Run: 0

Error: Invalid block: base fee not correct! Expected: '10', got: '0' (stRandom/randomStatetest153, fork: London, TrInfo: d: 0, g: 0, v: 0, TrData: )

--------
TestOutputHelper detected 1 errors during test execution!
/home/ia/dev/ethereum/retesteth/retesteth/TestOutputHelper.cpp(222): error: in 'GeneralStateTests/stRandom'

Date: 2022-11-11 08:02:47-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/evm/internal/t8ntool,cmd/evm: add missing EWASM interpreter flag to 'evm t8n' command

Date: 2022-11-11 08:04:57-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/evm/internal/t8ntool,cmd/evm: add missing EWASM interpreter flag to 'evm t8n' command

Date: 2022-11-11 08:04:57-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: DO NOT SKIP tests for Berlin,London when using -evmc.vm/wasm

Date: 2022-11-11 09:39:12-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: reimplement evmc.Storage* assignment matrix

Reference taken from
https://github.com/ethereum/evmc/blob/35112aed7b4e0fd2f54cf2e3272ba9117a0c6494/include/evmc/evmc.h

Date: 2022-11-11 09:39:56-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge branch 'evmc-v10-isaac' into merge/foundation-release/1.10.23-tests-generate-fixes-statefix-difficulty-test-v2-ripemd-tx-debug-no-ripemd-hack-evmc

meowsbits committed:
core/vm: still-failing attempt to install EVMC with revisions

Date: 2022-11-11 10:12:15-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: refactor/rewrite sstore handler for evmc host

This is still a WIP.
- native: all tests passing
- hera: all tests passing
  go test -v ./tests -run State -evmc.ewasm=/home/ia/go/src/github.com/ethereum/go-ethereum/build/_workspace/hera/build/src/libhera.so |& tee state.out
- evmone: many tests failing
  go test -v ./tests -run State -evmc.evm=/home/ia/go/src/github.com/ethereum/go-ethereum/build/_workspace/evmone/lib/libevmone.so |& tee state.out

Is this an evmone problem? An EVMC host-binding implementation problem?

Date: 2022-11-11 13:32:09-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/echainspec,cmd/faucet,cmd/geth,eth/tracers/internal/tracetest,ethclient,integration,node,params/confp/generic,convert_test,params/confp/tconvert,params/types/parity,tests: SA1019: package io/ioutil is deprecated

As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details. (staticcheck)

Date: 2022-11-11 14:08:10-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge pull request #509 from etclabscore/io-deprecated-go1.16

replace ioutil with os,io

meowsbits committed:
Merge branch 'master' into merge/foundation-release/1.10.23-tests-generate-fixes-statefix-difficulty-test-v2-ripemd-tx-debug-no-ripemd-hack-evmc

 Conflicts:
	tests/difficulty_mgen_test.go
	tests/state_mgen_test.go

meowsbits committed:
tests: [fixup/post-merge] s/ioutil.WriteFile/os.WriteFile/g

Date: 2022-12-06 07:41:32-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params: ExampleAllocPrint_Mainnet refers to unknown identifier: AllocPrint (govet)

Date: 2022-12-06 08:27:09-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: (was empty-branch) enable skip evmc tests at Berlin,London

Date: 2022-12-06 08:36:09-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: [lint/deadcode] test dirs defined but unused

Date: 2022-12-06 08:39:46-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: [lint] achieve perfection

Date: 2022-12-06 08:42:11-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/evm/testdata/13,cmd/evm/testdata/14,cmd/evm/testdata/19,cmd/evm/testdata/24,cmd/evm/testdata/25: git checkout foundation-master -- ./cmd/evm/testdata

These tests were failing because test expectations
(defined in json files) did not contain a 'currentBaseFee'
annotation that was expected.

Date: 2022-12-06 09:22:44-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/geth: bump geth console test timeout to 20s

These tests were failing because they were
timing out and getting killed.
I think the --dev mode key creation may account
for the problematic time spend.
This patch just increases the time the test
is willing to wait for its geth instance
to start up and print the logs the test is waiting
for.

Date: 2022-12-06 09:30:16-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: re-enable legacyStateTestDir

Date: 2022-12-06 11:03:11-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
.gitmodules,tests: move ETC testdata submodule to winsvega's ETC tests

Dimitry made us these tests, lets see if we can pass them.

Date: 2022-12-09 05:14:30-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: ensure context.Difficulty is assigned

context.Difficulty should be assigned by the env
in case 1559 is not enabled.
This makes sure that that happens.

Date: 2022-12-12 06:55:01-08:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
eth/filters,eth,eth/protocols/snap,graphql,params,rpc,trie: Merge branch 'foundation/v1.10.26' into merge/foundation-release/1.10.26

ziogaschr committed:
graphql: graphql: s/params.vars for InitialBaseFee

ziogaschr committed:
cmd/geth: cmd/geth: geth-test ipc was using a too-long (>108 char) path by default

meowsbits committed:
tests: update tests (incl. s/Merge/Merged/g fix), skip EOF1 tests

This core-geth does not yet have EOF1
implemented, so we add a skipLoad for that pattern.

Bump the submodule commit to a tests version that
fixes the Merge vs. Merged configuration naming issue.

Date: 2022-12-12 09:31:53-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: re-enable ETC legacy tests

Date: 2022-12-12 10:17:27-08:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
tests: .github/workflows,.gitmodules,Makefile,build,cmd/echainspec,cmd/evm/internal/t8ntool,cmd/evm,cmd/evm/testdata/13,cmd/evm/testdata/14,cmd/evm/testdata/19,cmd/evm/testdata/24,cmd/evm/testdata/25,cmd/faucet,cmd/geth,cmd/utils,consensus/ethash,core,core/state,core,core/types,core/vm,eth/ethconfig,eth/tracers/internal/tracetest,ethclient,go.mod,go.sum,integration,node,params,params/confp/generic,convert_test,params/confp/tconvert,params/coregeth.json.d,params/types/genesisT,params/types/parity,requirements-mkdocs.txt,tests: Merge branch 'merge/foundation-release/1.10.23-tests-generate-fixes-statefix-difficulty-test-v2-ripemd-tx-debug-no-ripemd-hack-evmc-dimitrys-etc-tests' of github.com:etclabscore/core-geth into merge/foundation-release/1.10.26

# Conflicts:
#	cmd/geth/consolecmd_cg_test.go

meowsbits committed:
params/types/coregeth,tests: fix GrayGlacier difficulty test

Only the testdata-etc suite has this
sub-suite of tests. (Because our tests/testdata
submodule may be outdated?)

Anyways, the chain config for the GrayGlacier fork
was not-existing, and the configurator
method (EIP5133) for core-geth was not configuring
the difficulty bomb delay table right.

ALSO, NOTE the json field name fix
s/eip5133FBloc/eip5133FBlock/

Date: 2022-12-19 09:55:55-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile: problem: tests are timing out on Github CI

See https://github.com/etclabscore/core-geth/actions/runs/3696522572/jobs/6260330266

solution: bump test timeout from default=10m to 20m

Date: 2022-12-19 09:56:57-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
build: problem: lint step timed out on Github CI

solution: bump lint timeout option to 5m

Date: 2022-12-19 10:05:02-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
.gitmodules: move ETH xclient tests submodule back to ethereum/tests

This had been moved to the etclabscore/tests fork
because at the time, ETC xclient tests were
living at this repo, too.
But now we've moved the ETC xclient tests to their
own suite with their own repo (etclabscore/tests@etcversion_filled)
and those tests have been generated via retesteth
by Dimitry.
So I want the original xclient suite (with only ETH)
tests to treat ethereum/tests as the upstream origin.

Date: 2022-12-19 10:13:58-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits commented on .gitmodules:
```diff
@@ -1,11 +1,11 @@
 [submodule "tests-etc"]
 	path = tests/testdata-etc
-	url = https://github.com/etclabscore/tests-etc
+	url = https://github.com/etclabscore/tests
```
This should have `shallow = true`. The repo is big and the story is long.

ziogaschr replied:
```suggestion
	url = https://github.com/etclabscore/tests
	shallow = true
```

@meowsbits it's your commit. Accept it

ziogaschr replied:
Resolved by [460091f](https://github.com/etclabscore/core-geth/pull/510/commits/460091f7268e2606cab870c43103ac410cda26ec)

meowsbits commented on Makefile:
```diff
@@ -48,24 +48,16 @@ test-coregeth: \
-
 # Test EVMC support against various external interpreters.
-test-evmc: hera ssvm evmone aleth-interpreter
+test-evmc: hera evmone
```
:notebook: EVMC support/development tooling for `aleth` and `ssvm` is dropped because they don't have an EVMCv10 version yet.

ziogaschr replied:
So you say that we can bring them soon with #507?

meowsbits commented on cmd/evm/internal/t8ntool/transition.go:
```diff
@@ -178,9 +179,20 @@ func Transition(ctx *cli.Context) error {
+	if vmConfig.EWASMInterpreter != "" {
+		vm.InitEVMCEwasm(vmConfig.EWASMInterpreter)
+	}
+
```
:notebook: Support for EVMC is added to `t8ntool` (used normally by `retesteth` for filling xclient tests). So `retesteth` can fill tests with moar EVMs.

meowsbits commented on cmd/evm/staterunner.go:
```diff
@@ -113,7 +118,16 @@ func stateTestCmd(ctx *cli.Context) error {
+	if cfg.EWASMInterpreter != "" {
+		vm.InitEVMCEwasm(cfg.EWASMInterpreter)
+	}
+
```
:notebook: xclient state tests can also be run with external EVMs.

meowsbits commented on cmd/faucet/faucet.go:
```diff
@@ -184,7 +183,7 @@ func parseChainFlags() (gs *genesisT.Genesis, bs string, netid uint64) {
 	// allow overrides
 	if *genesisFlag != "" {
-		blob, err := ioutil.ReadFile(*genesisFlag)
+		blob, err := os.ReadFile(*genesisFlag)
```
:notebook: `ioutil` deprecated; use `os` methods instead.

meowsbits commented on cmd/geth/consolecmd_cg_test.go:
```diff
@@ -97,11 +101,23 @@ func TestGethFailureToLaunch(t *testing.T) {
-	semiPersistentDatadir := filepath.Join(os.TempDir(), fmt.Sprintf("geth-startup-logs-test-%d", time.Now().Unix()))
+	// WATCHOUT: For Unix-based operating systems, you're going to have problems if the IPC endpoint is
+	// longer than ___ characters.
+	semiPersistentDatadir := filepath.Join(os.TempDir(), fmt.Sprintf("geth-test-%x", randomStr(4)))
```
:speech_balloon:  @ziogaschr Looks familiar... :wink: this you?

ziogaschr replied:
Yes man, copied your hack. :)

meowsbits commented on core/state/journal.go:
```diff
@@ -75,6 +75,7 @@ func (j *journal) revert(statedb *StateDB, snapshot int) {
 // dirty explicitly sets an address to dirty, even if the change entries would
 // otherwise suggest it as clean. This method is an ugly hack to handle the RIPEMD
 // precompile consensus exception.
+// meowsbits-ripemd
```
- [ ] Remove these developer hints.

ziogaschr replied:
Resolved by ca5dd0498f1ee4c7469f1c5172be50df0108f2c7

meowsbits commented on core/vm/evm.go:
```diff
@@ -22,7 +22,7 @@ import (
 	"time"
 
-	"github.com/ethereum/evmc/v7/bindings/go/evmc"
+	"github.com/ethereum/evmc/v10/bindings/go/evmc"
```
:notebook: Bumps EVMC API v7 -> v10.

meowsbits commented on core/vm/evmc.go:
```diff
@@ -26,7 +26,7 @@ import (
 	"sync"
 
-	"github.com/ethereum/evmc/v7/bindings/go/evmc"
+	"github.com/ethereum/evmc/v10/bindings/go/evmc"
```
I haven't yet revisited testing for this. IIRC last time I touched these, there were still some tests failing. See `make test-evmc`.

meowsbits commented on core/vm/evmc.go:
```diff
@@ -129,77 +147,297 @@ func (host *hostContext) GetStorage(addr evmc.Address, evmcKey evmc.Hash) evmc.H
-	}
 
-	var current, original uint256.Int
+	var current, original = new(uint256.Int), new(uint256.Int)
```
:spiral_notepad: Pointers now.

meowsbits commented on core/vm/evmc.go:
```diff
@@ -129,77 +147,297 @@ func (host *hostContext) GetStorage(addr evmc.Address, evmcKey evmc.Hash) evmc.H
-	if !hasNetStorageCostEIP {
-		status = evmc.StorageModified
-		if oldValue.IsZero() {
+	// defer func() {
```
- [ ] Remove dead comments.

ziogaschr replied:
Resolved by 7c9fc18bbe1888cc49464d1d254cce413ecf9d89

meowsbits commented on core/vm/evmc.go:
```diff
@@ -222,28 +460,34 @@ func (host *hostContext) GetCode(addr evmc.Address) []byte {
+		Number:    host.env.Context.BlockNumber.Int64(),
+		Timestamp: host.env.Context.Time.Int64(),
+		GasLimit:  int64(host.env.Context.GasLimit),
+		ChainID:   evmc.Hash(common.BigToHash(host.env.chainConfig.GetChainID())),
```
`Difficulty:` ... ?  (This field gets repurposed with PoS; it needs to be handled carefully.)

meowsbits commented on core/vm/evmc.go:
```diff
@@ -346,10 +597,12 @@ func getRevision(env *EVM) evmc.Revision {
-		panic("berlin is unsupported by EVMCv7")
-	case conf.IsEnabled(conf.GetEIP1884Transition, n):
+		return evmc.Berlin
+	case conf.IsEnabled(conf.GetEIP2200Transition, n):
```
:notebook: Note the new 'canary' features for forks.

meowsbits commented on params/types/coregeth/chain_config.go:
```diff
@@ -215,7 +215,8 @@ type CoreGethChainConfig struct {
 
-	EIP5133FBlock *big.Int `json:"eip5133FBloc,omitempty"`
+	EIP5133FBlock   *big.Int `json:"eip5133FBlock,omitempty"`
+	eip5133Inferred bool
```
Here's my fix for `GrayGlacier` xclient difficulty tests.

meowsbits commented on params/types/genesisT/genesis.go:
```diff
@@ -32,6 +32,9 @@ import (
 )
 
+//go:generate go run github.com/fjl/gencodec -type Genesis -field-override genesisSpecMarshaling -out gen_genesis.go
+//go:generate go run github.com/fjl/gencodec -type GenesisAccount -field-override genesisAccountMarshaling -out gen_genesis_account.go
```
What if we had a CI test that made sure developers don't forget to run this. Could just check no-change after run?

ziogaschr replied:
Great point @meowsbits

meowsbits commented on rpc/client.go:
```diff
@@ -31,6 +31,7 @@ import (
 )
 
 var (
+	ErrBadResult                 = errors.New("bad result in JSON-RPC response")
```
Why does the client give a bad result in the first place?

meowsbits commented on rpc/http.go:
```diff
@@ -173,6 +173,9 @@ func (c *Client) sendBatchHTTP(ctx context.Context, op *requestOp, msgs []*jsonr
 	}
+	if len(respmsgs) != len(msgs) {
+		return fmt.Errorf("batch has %d requests but response has %d: %w", len(msgs), len(respmsgs), ErrBadResult)
+	}
```
Like, why? Seriously?

ziogaschr replied:
It has to do with channel issues and locking. 
Check here  https://github.com/ethereum/go-ethereum/pull/26064

meowsbits commented on tests/init_mgen_test.go:
```diff
@@ -58,86 +54,86 @@ func (tm *testMatcher) walkFullName(t *testing.T, dir string, runTest interface{
 }
 
-// walk invokes its runTest argument for all subtests in the given directory.
+// // walk invokes its runTest argument for all subtests in the given directory.
```
- [ ] Remove dead code.

meowsbits commented on tests/init_test.go:
```diff
@@ -192,17 +191,17 @@ func (tm *testMatcher) findSkip(name string) (reason string, skipload bool) {
-	log.Println("using empty config", name)
-	return new(coregeth.CoreGethChainConfig), ""
-}
+// // findConfig returns the chain config matching defined patterns.
```
- [ ] Remove dead code.

meowsbits commented on tests/state_test.go:
```diff
@@ -75,7 +77,7 @@ func TestState(t *testing.T) {
 		// Berlin tests are not expected to pass for external EVMs, yet.
 		//
-		st.skipFork("^Berlin$") // ETH
+		st.skipFork("Berlin")   // ETH
```
These `skips` need removed; support provided by EVMCv10.

ziogaschr replied:
Does it cover both hera and evmone?

meowsbits replied:
Yes. EVMOne is EVM, Hera is EWASM.

meowsbits commented on tests/state_test_util.go:
```diff
@@ -318,6 +318,8 @@ func (t *StateTest) RunNoVerify(subtest StateSubtest, vmconfig vm.Config, snapsh
 		context.Random = &rnd
 		context.Difficulty = big.NewInt(0)
+	} else {
+		context.Difficulty = t.json.Env.Difficulty
```
I don't think -- off the top of my head -- this is necessary; there may be supporting logic in the `.genesis(fork)` call that sets this too.  `Difficulty` field, touchy touchy.

ziogaschr committed:
.gitmodules: .gitmodules: set shallow mode for tests-etc submodule

ziogaschr committed:
core/state: core/state: remove comment

ziogaschr committed:
core/vm: core/vm: remove comment

meowsbits committed:
tests: remove dead code (state,diff mgen artifacts)

This code was used in earlier version of
the test generation logic, which extended
some given xclient suite with ETC support
by writing the results of a test to a new file.

Difficulty tests at the time used NDJSON
as an additional encoding, but this is no longer.

Now, core-geth uses xclient tests generated by
retesteth (which uses core-geth as a filler client;
actually the t8ntool command provided by the repo).

Resolves

- https://github.com/etclabscore/core-geth/pull/510#discussion_r1052683037
- https://github.com/etclabscore/core-geth/pull/510#discussion_r1052683214

Date: 2022-12-20 06:28:24-08:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
.github/workflows: add check for “go generate” diff

Usually devs forget to update Marshalers that has been generated by “go generate”, this Action will just check if a diff exists and let the developers know on a PR basis

ziogaschr committed:
.github/workflows: .github/workflows: no need for submodules for go:generate

ziogaschr committed:
.github/workflows: .github/workflows: install devtools for “go:generate” deps

ziogaschr committed:
.github/workflows: use solc version “0.6.12”

ziogaschr committed:
.github/workflows: use solc version 0.6.12

ziogaschr committed:
.github/workflows: remove go-generate-check unneeded rm

meowsbits committed:
contracts/checkpointoracle,contracts/checkpointoracle/contract,params/types/genesisT: run 'go generate ./...' with partial success

A few things I learned along the way this evening.

- solc in path must be 0.6.0 or some mysterious upper
bound less than the latest available versions that I
also tried.
- solcjs is not solc. npm install solc installs solcjs
which is like solc but is not solc.

All of this partially fixes an error which looked like this.

go generate ./...
protoc-gen-go: no such flag -import_path
--go_out: protoc-gen-go: Plugin failed with status code 1.
accounts/usbwallet/trezor/trezor.go:45: running 'protoc': exit status 1
Error: Source file requires different compiler version (current compiler is 0.8.5+commit.a4f2e591.Linux.g++) - note that nightly builds are considered to be strictly less than the released version
 --> contract/oracle.sol:1:1:
    |
           1 | pragma solidity ^0.6.0;
         | ^^^^^^^^^^^^^^^^^^^^^^^

              contracts/checkpointoracle/oracle.go:20: running 'solc': exit status 1

The error now looks like this

go generate ./...
protoc-gen-go: no such flag -import_path
--go_out: protoc-gen-go: Plugin failed with status code 1.

Maybe we don't even want these changes. I don't know.
I just want the command to work for the sake of order
in the universe.

But seriously
- maybe we should upgrade the Solidity version pragma too/instead?
- maybe proto-gen-go actually wants a different version than 'latest',
  (which is what 'make devtools' installs).
- maybe we report the issue at ethereum. their trezor.go looks same
  same.

Goodnight, to dream of green lights.

Date: 2022-12-20 19:40:04-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/genesisT: edit the document which shall not be edited (add config type switch)

Date: 2023-01-10 08:11:37-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/devp2p/internal/ethtest: these tests fails spuriously; solve with 1s sleep

They depend on the output of a geth instance.
Geth can take a while to start up, and assertions
about geth logs should normally use some kind of a hook
to decide when its ok to check the logs (eg startup complete).

Date: 2023-01-10 09:48:43-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: checkout master@b909e85

Revering the EVMCv10 upgrade.
I can't get all tests to pass with EVM One.

Date: 2023-01-10 10:36:57-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: skip fork Merged for EVMC test suites

Date: 2023-01-10 10:38:13-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm,go.mod,go.sum: revert ethereum/evmc v10 -> v7

Date: 2023-01-10 10:40:11-08:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/devp2p/internal/ethtest: register runGeth fn as a t.Helper

These tests keep failing... sometimes.
This is a shot in the dark.

Date: 2023-01-10 11:54:56-08:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
.github/workflows: .github/workflows: add whitelisted generated files in go-generate-check

We currently have a modified generated file which will cause this check to always file, until we actually fix the generated code, we decided to whitelist this file.

The file is: `params/types/genesisT/gen_genesis.go`

ziogaschr committed:
.github/workflows: .github/workflows: fix go-generate-check issues

- skip protobuf installation as trezor code generation requires special handling
- whitelist `params/types/genesisT/gen_genesis.go` which is custom modified

ziogaschr committed:
.github/workflows: .github/workflows: .github/workflows: fix emitting a failure in go-generate-check

ziogaschr commented on .github/workflows/go-generate-check.yml:
```diff
@@ -0,0 +1,50 @@
+        id: go-generate
+        run: |
+          # ignore generating trezor package based on this comment:
+          # https://github.com/ethereum/go-ethereum/blob/master/accounts/usbwallet/trezor/trezor.go#L21-L43
```
@meowsbits please have a look at the reason I decide to skip the trezor generating code which also requires the protobuffer.

I assume that this code if/when needs to be generated we need to take extra manual care, for this reason I suggest to net even check it for differences

ziogaschr commented on .github/workflows/go-generate-check.yml:
```diff
@@ -0,0 +1,50 @@
+        id: skip-files
+        run: |
+          # ignore `gen_genesis.go` file which has custom modifications for unmarshaling the various genesis formats
+          git checkout params/types/genesisT/gen_genesis.go
```
@meowsbits here I suggest ignoring the `gen_genesis.go` file which has custom modification for core-geth support of different genesis flavours for each client.

ziogaschr committed:
.github/workflows: .github/workflows: use solc@0.6.0 in go-generate-check for making checkpointoracle

meowsbits committed:
.github/workflows: refactor exceptions and ignores to env vars

I think this helps with readability.

Date: 2023-01-11 09:03:55-08:00
Signed-off-by: meows <b5c6@protonmail.com>

whilei committed:
Merge branch 'merge/foundation-release/1.10.26' of meowsbits.github.com:etclabscore/core-geth into merge/foundation-release/1.10.26

ziogaschr committed:
.github/workflows: .github/workflows: fix env vars list parsing in go-generate-check.yml

meowsbits committed:
Merge pull request #512 from etclabscore/merge/foundation-release/1.10.26-isaac-no-evmc

Merge/foundation release/1.10.26 <- Remove EVMCv10 upgrade, minor fixes

ziogaschr committed:
.github/workflows: .github/workflows: .github/workflows: .github/workflows: go-generate-check changed logic for whitelisting changes in generated files directly

The intention of this commit is to not break/change the logic for developers when coding.
This script serves as a warning notification if a developer forgets to generate those files using gencodec.

On specific cases where generated files are modified, we revert the modifications and compare the files so as they will show up any changes in the original files.

The developer then will find out the correct way to resolve generating the new files and keeping the modifications intact if needed.
